### DISCUSSION TITLE
How to fire a mutation from another component?

### DISCUSSION THREAD

**Discussion by shinobu-uwu** [TanStack/query#8057] [Q&A]
Hello

I have one parent component that is where data fetching happens, with `useQuery`, what I'd like to do is: define a mutation in that component, and fire it from a child. I've tried to use `useMutationState` and select the `mutation.execute` field, but `useMutationState` always returns an empty array, even with a key set and it being the only filter.

Also, I'm not using the mutation in my parent component at all, I'd just like to define it there and the children are the one to fire said mutation, prop drilling works, but it's not a very good solution

**Comment by TkDodo** [comment]
It's best to define things where you actually need them. Why would you want it in a parent that doesn't need it? It's like defining state in the root component. Yes you can use context or prop drilling to pass it down but as you said, that's not a very good solution.

**Reply by shinobu-uwu** [reply]
Actually I think I'm not understand mutations correctly, I was in a hurry to turn in the task when I asked that, so I just went with prop drilling and it's working, I'll take some time to read more in depth the docs later, but what I have now is:

Parent component that fetches the data and builds the list entries that is extracted to another component. The children don't have, or shouldn't have though it's possible, access to the filters of the query, so I think it's not possible to `useQueryClient` to get the query, or is it? My key is something like `queryKey: ['entries', debounceFilter, sortBy, tenant]`. I'm trying to do optimistic updates on delete operation and the delete button is in the child component.
